dask/base.py

```diff
@@ -249,7 +249,9 @@ def normalize_function(func):
         return (normalize_function(func.func), func.args, kws)
     else:
         try:
-            return pickle.dumps(func, protocol=0)
+            result = pickle.dumps(func, protocol=0)
+            assert b'__main__' not in result  # abort on dynamic functions
```

Comment:
Using an assert for control-flow here will mean that the (rare) users who use `python -O` will [see different behavior](https://docs.python.org/3/reference/simple_stmts.html?highlight=assert#the-assert-statement).